### PR TITLE
fix - drip rate group debt when calculating writeoff value

### PR DESCRIPTION
**Pull request by ilinzweilin**
- when calculating NAV: drip rate group debt for rwriteOff rate groups
- file writeOff rate groups on NAV creation

### CODE REVIEW

File: src/borrower/feed/navfeed.sol [hunk 1]
```diff
@@ -76,10 +76,15 @@ contract NAVFeed is BaseNFTFeed, Interest, Buckets, FixedPoint {
         recoveryRatePD[4] = Fixed27(ONE);
 
         /// Overdue loans (= loans that were not repaid by the maturityDate) are moved to write Offs
-        // 60% -> 40% write off
-        writeOffs[0] = WriteOff(WRITE_OFF_PHASE_A, Fixed27(6 * 10**26));
-        // 80% -> 20% write off
-        writeOffs[1] = WriteOff(WRITE_OFF_PHASE_B, Fixed27(8 * 10**26));
+        // 6% rate -> 60% write off
+        setWriteOff(0, WRITE_OFF_PHASE_A, uint(1000000674400000000000000000), 6 * 10**26);
```
Comment (manuelpolzhofer) [inline src/borrower/feed/navfeed.sol:80]: Can you add in the comments how high the interest rate is?

File: src/borrower/feed/navfeed.sol [hunk 2]
```diff
@@ -256,11 +261,10 @@ contract NAVFeed is BaseNFTFeed, Interest, Buckets, FixedPoint {
...
         // calculates the NAV for ongoing loans with a maturityDate date in the future
         uint nav_ = calcTotalDiscount();
-
-        // include ovedue assets (maturity date passed) to the current NAV calculation
-        for (uint i = 0; i < writeOffs.length; i++) {
-            (uint pie, uint chi, , ,) = pile.rates(writeOffs[i].rateGroup);
-            nav_ = safeAdd(nav_, rmul(rmul(pie, chi), writeOffs[i].percentage.value));
+        // include ovedue assets to the current NAV calculation
+        for (uint i = 0; i < writeOffs.length; i++) {       
+            // multiply writeOffGroupDebt with the writeOff ratio
```
Comment (manuelpolzhofer) [inline src/borrower/feed/navfeed.sol:266]: Is it actually a ratio?

File: src/borrower/feed/navfeed.sol [hunk 3]
```diff
@@ -256,11 +261,10 @@ contract NAVFeed is BaseNFTFeed, Interest, Buckets, FixedPoint {
...
+            nav_ = safeAdd(nav_, rmul(pile.rateDebt(writeOffs[i].rateGroup), writeOffs[i].percentage.value));
```
Comment (manuelpolzhofer) [inline src/borrower/feed/navfeed.sol:267]: 👍

File: src/borrower/feed/nftfeed.sol [hunk 4]
```diff
@@ -31,8 +31,9 @@ contract PileLike {
     function changeRate(uint loan, uint newRate) public;
     function loanRates(uint loan) public returns (uint);
     function file(bytes32, uint, uint) public;
-    function rates(uint rate) public view returns(uint, uint, uint ,uint48, uint);
-    function total() public view returns(uint);
+    function rates(uint rate) public view returns (uint, uint, uint ,uint48, uint);
+    function total() public view returns (uint);
```
Comment (manuelpolzhofer) [inline src/borrower/feed/nftfeed.sol:35]: Where are we using this two?

**Review by manuelpolzhofer — APPROVED** [review]

File: src/borrower/feed/navfeed.sol (hunk 2 above)
Comment (ilinzweilin) [reply src/borrower/feed/navfeed.sol:266]: it's rather a rate

File: src/borrower/feed/nftfeed.sol (hunk 4 above)
Comment (ilinzweilin) [reply src/borrower/feed/nftfeed.sol:35]: for the fv calculation & totalValue in nftFeed

**Pull request merged**